hw3/3-5/src/pages/Search.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect } from "react";
+import Container from "react-bootstrap/Container";
+import Form from "react-bootstrap/Form";
+import InputGroup from "react-bootstrap/InputGroup";
+import "./styles/search.css";
+
+const Search = () => {
+  const [thrones, setThrones] = useState([]);
+  const [allThrones, setAllThrones] = useState([]);
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const handleSearchInput = (event) => {
+    const newSearchTerm = event.target.value;
+    setSearchTerm(newSearchTerm);
+    handleSearch(newSearchTerm);
+  };
+
+  const handleSearch = (inputSearchTerm) => {
+    if (inputSearchTerm) {
+      const searchResults = allThrones.filter((character) =>
+        character.fullName.toLowerCase().includes(inputSearchTerm.toLowerCase())
+      );
+      setThrones(searchResults);
+    }
+  };
+
+  const fetchThrones = async (url) => {
+    try {
+      const response = await fetch(url);
+      const data = await response.json();
+      setAllThrones(data);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  useEffect(() => {
+    fetchThrones("https://thronesapi.com/api/v2/Characters");
```

Comment:
same as above, the URL doesn't change so we should move it to a common file

Reply:
I got you ... all these in a common config files kind of